src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -43,27 +88,344 @@ public async Task<IEnumerable<TimeSeriesValue>> GetAsync(string timeseries, Date
 
             Session.IncrementRequestCount();
 
-            var details = await Session.Operations.SendAsync(
-                 new GetTimeSeriesOperation(DocId, timeseries, from, to), Session.SessionInfo, token: token)
-                 .ConfigureAwait(false);
-            
-            //TODO: if value exists, merge new values
+            details = await Session.Operations.SendAsync(
+                    new GetTimeSeriesOperation(DocId, timeseries, from, to), Session.SessionInfo, token: token)
+                .ConfigureAwait(false);
 
             if (Session.NoTracking == false)
             {
-                if (Session.TimeSeriesByDocId.TryGetValue(DocId, out cache) == false)
+                if (Session.TimeSeriesByDocId.TryGetValue(DocId, out cache) == false)  
+                {
+                    Session.TimeSeriesByDocId[DocId] = cache = new Dictionary<string, List<TimeSeriesRange>>(StringComparer.OrdinalIgnoreCase);
+                }
+
+                cache[timeseries] = new List<TimeSeriesRange>
+                {
+                    details.Values[timeseries][0]
+                };
+                
+            }
+
+            return details.Values[timeseries][0].Values;
+
+            TimeSeriesValue[] MergeRangesAndResults(int fromRangeIndex, int toRangeIndex, List<TimeSeriesRange> resultFromServer, out IEnumerable<TimeSeriesValue> resultToUser)
```

Comment:
I'm having a hard time going through this function.
Can you add some comments that explain the reasoning?

Reply:
added some comments